src/main/java/rx/subscriptions/BooleanSubscription.java

```diff
@@ -28,9 +28,7 @@
 public final class BooleanSubscription implements Subscription {
 
     private final Action0 action;
```

Comment:
Generally, I'm not satisfied with the retention of `Action0` in this class. Instead of `unsubscribed` you could introduce an `AtomicReference<Action0>` plus a private static final empty action (don't use `Actions.empty()` !). [Here](https://github.com/ReactiveX/RxJava/blob/2.x/src/main/java/io/reactivex/disposables/BooleanDisposable.java) is an example for this kind of atomics.